app/models/my_module.rb

```diff
@@ -373,12 +373,22 @@ def toggle_tab(tab)
 
       if shown_tabs.include?(tab)
         if can_uncheck
+          my_module_widgets.where(widget_type: MyModuleWidget.widget_types[tab])
```

Comment:
I would add comment here just so we know why is this so and that it's a dirty "hack" that might be removed in future.